pyanaconda/kickstart_dispatcher.py

```diff
@@ -0,0 +1,355 @@
+        :type process: bool
+
+        :return: list of filtered elements
+        :rtype: list(KickstartElement)
+        """
+        elements = super().get_elements(commands, sections, addons)
+        if process:
+            self._processed_elements.update(elements)
+        return elements
+
+    @property
+    def unprocessed_elements(self):
+        """List of all elements not tracked as processed.
+
+        :return: list of all elements not tracked as processed
+        :rtype: list(KickstartElement)
+        """
+        return [element for element in self._elements
+                if element not in self._processed_elements]
+
+
+class StoreSection(Section):
+    """Section for storing section content and header line references.
+
+    Similarly as NullSection defines a section that parser will recognize (ie
+    will not raise an error). The section will pass itself to an object for storing
+    sections if supplied.
+    """
+
+    allLines = True
+
+    def __init__(self, *args, **kwargs):
+        """Create a new StoreSection instance.
+
+        You must pass a sectionOpen parameter (including a leading '%') for the
+        section to make it valid but just ignored. If you want to store the
+        content, supply a store argument.
+
+        Required kwargs:
+        sectionOpen - section name, including '%' starting character
+
+        Optional kwargs:
+        store - an instance of an object for storing the section
+                (SplitKickstartParser) which must provide
+                add_section(StoreSection) method
+        """
+        super().__init__(*args, **kwargs)
+        self.sectionOpen = kwargs.get("sectionOpen")
+        self._store = kwargs.get("store", None)
+        self.header_lineno = 0
+        self.args = []
+        self._body = []
+
+    def handleHeader(self, lineno, args):
+        self.header_lineno = lineno
+        self.args = args
+
+    def handleLine(self, line):
+        self._body.append(line)
+
+    def finalize(self):
+        if self._store is not None:
+            self._store.add_section(self)
+        self.header_lineno = 0
+        self.args = []
+        self._body = []
+
+    def __str__(self):
+        body = "".join(self._body)
+        if body:
+            s = "{}\n{}%end\n".format(" ".join(self.args), body)
+        else:
+            s = "{}\n%end\n".format(" ".join(self.args))
+        return s
+
+
+class SplitKickstartParser(KickstartParser):
+    """Kickstart parser for storing kickstart elements.
+
+    Stores kickstart elements (commands, sections, addons) with their line
+    number and file name references to kickstart file.
+    Does not do any actual command or section parsing (ie command syntax
+    checking).
+
+    :raises KickstartParseError: on invalid section
+    :raises KickstartError: on missing %include unless instantiated with
+                            missing_include_is_fatal=False
+    """
```

Comment:
I'm able to imagine this as a library and this will be the interface of a library. I guess nothing else that this should be used outside in  the code. So why not to have this in `__init__.py` fine in different folder to separate all the implementation details (other classes) from developer when he/she wants to use this.

Reply:
I see the interface as one layer above.
This is just modified KickstartParser, although it seems a bit like providing THE interface.